x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/trainedmodel/tree/Tree.java

```diff
@@ -253,6 +253,9 @@ public static Builder builder() {
 
     @Override
     public void validate() {
+        if (featureNames.isEmpty()) {
+            throw ExceptionsHelper.badRequestException("[feature_names] must not be empty for tree model");
```

Comment:
Would it make sense to use `FEATURE_NAMES.getPreferredName()` here rather than literal?